### PR TITLE
Pycbc waveform generation

### PR DESCRIPTION
**Pull request by wbenoit26**
This PR changes our waveform generation method from `bilby` to `pycbc`. Along the way, it also:

- Creates a new `cosmology` library to store whatever cosmological calculations we need. In particular, calculation of astrophysical volume now lives here, and there's now a `DEFAULT_COSMOLOGY` variable for standardization purposes.
- Updates `rejection.py` to generate waveforms using the `from_parameter` functionality
- Renames/rearranges some of the parameter ledger objects in a way that I _think_ makes more sense. I tried to copy a bit from how `pycbc` organizes their `ParameterList`s
- Changes the `theta_jn` parameter distribution in `priors.py` to `inclination` to match what `pycbc` expects
- Changed `phi_jl` and `phi_12` parameter distributions to be distributions over individual azimuthal angles of each object to allow for easier conversion to spin parameters that `pycbc` expects

To-do:
- [x] Test whether all of this works

Training and validation waveform generation ran, and testing waveforms are running now. The training run is [here](https://wandb.ai/ml4gw/aframe/runs/ag5s3vqu?nw=nwuserbenoi090).

@EthanMarx there's still some work to be done on this, but I'd be interested in getting your feedback on the structure.

### CODE REVIEW

**Comment by EthanMarx** [comment]
This is awesome and long overdue. I'll take a look at the details in a bit, but it may make sense to just add the `cosmology` module to the `utils` library, especially since it's a single module with limited functionality currently. Will also keep our dependencies from getting too bloated. I'll leave it up to you though.

File: libs/architectures/pyproject.toml [hunk 1]
```diff
@@ -14,8 +14,7 @@ h5py = "^3.9.0"
 [[tool.poetry.source]]
 name = "torch"
 url = "https://download.pytorch.org/whl/cu118"
-secondary = true
-default = false
+priority = "explicit"
```
Comment (EthanMarx) [inline libs/architectures/pyproject.toml:17]: I'm guessing this syntax got depreated?

Comment (wbenoit26) [reply libs/architectures/pyproject.toml:17]: Yeah, this is the syntax as of 1.5. It also stops all of those spurious "authorization" warnings, which is nice

**Comment by wbenoit26** [comment]
I agree about the cosmology lib. I don't foresee us having many more functions that would fit in there. I'll move it into utils

File: projects/plots/apptainer.def [hunk 2]
```diff
@@ -44,3 +44,4 @@ apt-get clean
 cd /opt/aframe/projects/plots
 pip install -r requirements.txt --no-deps --no-cache-dir
 pip install -e . --no-deps --no-cache-dir
+pip install pytest
```
Comment (EthanMarx) [inline projects/plots/apptainer.def:47]: Should probably just add this as a dev dep in the `pyproject.toml`

Comment (wbenoit26) [reply projects/plots/apptainer.def:47]: Oh, yup, I thought it was already there. I think it needs to be in the apptainer file though - we have it for `train`, `export`, and `infer`.

File: .github/workflows/lib-tests.yaml [hunk 3]
```diff
@@ -58,6 +58,9 @@ jobs:
         env:
           test_dir: /github/workspace/libs/${{ matrix.library }}
         run: |
+            apt-get update
```
Comment (EthanMarx) [inline .github/workflows/lib-tests.yaml:61]: what's the reason for needing this install?

Comment (wbenoit26) [reply .github/workflows/lib-tests.yaml:61]: pycbc depends on ligo-segments, which required gcc to be installed for whatever reason

Comment (EthanMarx) [reply .github/workflows/lib-tests.yaml:61]: Got it, does this break our local poetry installs of things that depend on these libs?

Comment (wbenoit26) [reply .github/workflows/lib-tests.yaml:61]: No, those still work fine, though I guess it's something to look out for

**Comment by wbenoit26** [comment]
The testing finished up and the SV looks bad: [https://ldas-jobs.ligo-wa.caltech.edu/~william.benoit/pycbc_waveform_sv.html](https://ldas-jobs.ligo-wa.caltech.edu/~william.benoit/pycbc_waveform_sv.html). The training curves look good, and some quick checks of the testing waveforms look good, so I'm not sure what the issue is. I'll look into it some more.

**Comment by wbenoit26** [comment]
@EthanMarx I'm thinking that this PR might be a good place to start maintaining a `dev` branch

File: .github/workflows/lib-tests.yaml [hunk 4]
```diff
@@ -58,6 +58,8 @@ jobs:
...
```
Comment (EthanMarx) [inline .github/workflows/lib-tests.yaml:61]: @wbenoit26 I think doing `sudo apt-get update` (and adding `sudo` to the system install commands) might help your problem

Comment (wbenoit26) [reply .github/workflows/lib-tests.yaml:61]: I found the problem after trying all of this yesterday - it's actually that GitHub CLI's GPG key [expired](https://github.com/cli/cli/issues/9569) on September 6th, so running `update` fails to find the correct signature.

The container we use automatically uses the most recent GPG key, so this should be fixed by re-building the image. I filed an issue on the `pyaction` repo yesterday, and the owner said he'd take a look at it today.

Comment (EthanMarx) [reply .github/workflows/lib-tests.yaml:61]: okay cool. Is this a result of having to install `gcc` for pycbc?

Comment (wbenoit26) [reply .github/workflows/lib-tests.yaml:61]: Yeah, entirely. I wouldn't need any of the `apt-get` steps if it weren't for that, which is a little annoying.

**Comment by wbenoit26** [comment]
@EthanMarx I'm going to open up a test PR so I can debug some of the container building issues that are happening here. I may also close this PR and create a new one that's a little more focused.

**Comment by EthanMarx** [comment]
Yeah - splitting this up into smaller PRs would be great. What's the core of the issue?  Can you reproduce the container build failure locally?

**Pull request closed without merging**